

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" />
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  
  <title>count-alleles &mdash; HATCHet 0.3.1 documentation</title>
  

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />

  
  

  
  

  

  
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="index.html" class="icon icon-home"> HATCHet
          

          
          </a>

          
            
            
              <div class="version">
                0.3.1
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul>
<li class="toctree-l1"><a class="reference internal" href="README.html">HATCHet</a></li>
</ul>
<ul>
<li class="toctree-l1"><a class="reference internal" href="modules.html">Modules</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">HATCHet</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          

















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html" class="icon icon-home"></a> &raquo;</li>
        
      <li>count-alleles</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
          
            <a href="_sources/doc_count_alleles.md.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="count-alleles">
<h1>count-alleles<a class="headerlink" href="#count-alleles" title="Permalink to this headline">¶</a></h1>
<p>This step of HATCHet calls heterozygous germline SNPs from the matched-normal sample and counts the number of reads covering both the alleles of each identified heterozgyous SNP in every tumor sample.</p>
<div class="section" id="input">
<h2>Input<a class="headerlink" href="#input" title="Permalink to this headline">¶</a></h2>
<p>count-alleles takes in input sorted and indexed BAM files for multiple tumor samples from the same patient, a sorted and index BAM file from a matched-normal sample, and a indexed human reference genome.</p>
<table class="colwidths-auto docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Name</p></th>
<th class="head"><p>Description</p></th>
<th class="head"><p>Usage</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">-T</span></code>, <code class="docutils literal notranslate"><span class="pre">--tumors</span></code></p></td>
<td><p>A white-space separated list of sorted-indexed BAM files</p></td>
<td><p>The tumor samples from the same patient that are jointly analyzed by HATCHet</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">-N</span></code>, <code class="docutils literal notranslate"><span class="pre">--normal</span></code></p></td>
<td><p>A sorted-indexed BAM file</p></td>
<td><p>The matched normal sample for the same patient</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">-r</span></code>, <code class="docutils literal notranslate"><span class="pre">--reference</span></code></p></td>
<td><p>A FASTA file</p></td>
<td><p>The human reference genome used for germline variant calling</p></td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="output">
<h2>Output<a class="headerlink" href="#output" title="Permalink to this headline">¶</a></h2>
<p>count-alleles produces three tab-separated files: the first contains the read counts for every genomic bin in every tumor sample, the second contains the read counts for every genomic bin the matched-normal sample, and the third contains a list of the genomic positions that have been identified as germline heterozygous SNPs in the matched-normal sample.</p>
<table class="colwidths-auto docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Name</p></th>
<th class="head"><p>Description</p></th>
<th class="head"><p>Format</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">-O</span></code>, <code class="docutils literal notranslate"><span class="pre">--outputnormal</span></code></p></td>
<td><p>The output file for the read counts from matched-normal sample</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">#SAMPLE</span>&#160; <span class="pre">CHR</span>&#160; <span class="pre">POS</span>&#160; <span class="pre">REF_COUNT</span>&#160; <span class="pre">ALT_COUNT</span></code></p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">-o</span></code>, <code class="docutils literal notranslate"><span class="pre">--outputtumors</span></code></p></td>
<td><p>The output file for the read counts from the tumor samples</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">#SAMPLE</span>&#160; <span class="pre">CHR</span>&#160; <span class="pre">POS</span>&#160; <span class="pre">REF_COUNT</span>&#160; <span class="pre">ALT_COUNT</span></code></p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">-l</span></code>, <code class="docutils literal notranslate"><span class="pre">--outputsnps</span></code></p></td>
<td><p>the output file for the list of identified heterozygous germline SNPs</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">#CHR</span> <span class="pre">POS</span></code></p></td>
</tr>
</tbody>
</table>
<p>The format fields are described in the following.</p>
<table class="colwidths-auto docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Field</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">SAMPLE</span></code></p></td>
<td><p>Name of a sample</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">CHR</span></code></p></td>
<td><p>Name of the chromosome</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">POS</span></code></p></td>
<td><p>Genomic position in <code class="docutils literal notranslate"><span class="pre">CHR</span></code></p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">REF_COUNT</span></code></p></td>
<td><p>Number of reads harboring reference allele in <code class="docutils literal notranslate"><span class="pre">POS</span></code></p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">ALT_COUNT</span></code></p></td>
<td><p>Number of reads harboring alternate allele in <code class="docutils literal notranslate"><span class="pre">POS</span></code></p></td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="main-parameters">
<h2>Main parameters<a class="headerlink" href="#main-parameters" title="Permalink to this headline">¶</a></h2>
<p>count-alleles has some main parameters; the main values of these parameters allow to deal with most of datasets, but their values can be changed or tuned to accommodate the features of special datasets.</p>
<table class="colwidths-auto docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Name</p></th>
<th class="head"><p>Description</p></th>
<th class="head"><p>Usage</p></th>
<th class="head"><p>Default</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">-S</span></code>, <code class="docutils literal notranslate"><span class="pre">--samples</span></code></p></td>
<td><p>White-space separater list of a names</p></td>
<td><p>The first name is used for the matched-normal sample, while the others are for the tumor samples and they match the same order of the corresponding BAM files</p></td>
<td><p>File names are used</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">-st</span></code>, <code class="docutils literal notranslate"><span class="pre">--samtools</span></code></p></td>
<td><p>Path to <code class="docutils literal notranslate"><span class="pre">bin</span></code> directory of SAMtools</p></td>
<td><p>The path to this direcoty needs to be specified when it is not included in <code class="docutils literal notranslate"><span class="pre">$PATH</span></code></p></td>
<td><p>Path is expected in the enviroment variable <code class="docutils literal notranslate"><span class="pre">$PATH</span></code></p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">-bt</span></code>, <code class="docutils literal notranslate"><span class="pre">--bcftools</span></code></p></td>
<td><p>Path to <code class="docutils literal notranslate"><span class="pre">bin</span></code> directory of BCFtools</p></td>
<td><p>The path to this direcoty needs to be specified when it is not included in <code class="docutils literal notranslate"><span class="pre">$PATH</span></code></p></td>
<td><p>Path is expected in the enviroment variable <code class="docutils literal notranslate"><span class="pre">$PATH</span></code></p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">-c</span></code>, <code class="docutils literal notranslate"><span class="pre">--mincov</span></code></p></td>
<td><p>Minimum coverage</p></td>
<td><p>Minimum number of reads that have to cover a variant to be called, the value can be increased when considering a dataset with high depth (&gt;60x)</p></td>
<td><p>8</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">-C</span></code>, <code class="docutils literal notranslate"><span class="pre">--maxcov</span></code></p></td>
<td><p>Maximum coverage</p></td>
<td><p>Maximum number of reads that have to cover a variant to be called, the typically suggested value should be twice higher than expected coverage to avoid sequencing and mapping artifacts</p></td>
<td><p>300</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">-j</span></code>, <code class="docutils literal notranslate"><span class="pre">--processes</span></code></p></td>
<td><p>Number of parallele jobs</p></td>
<td><p>Parallel jobs are used to consider the chromosomes in different samples on parallel. The higher the number the better the running time</p></td>
<td><p>22</p></td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="optional-parameters">
<h2>Optional parameters<a class="headerlink" href="#optional-parameters" title="Permalink to this headline">¶</a></h2>
<p>count-alleles has some optional parameters; changes in the default values of these parameters are not expected to have a significant impact but they can be tuned to better fit the given data.</p>
<table class="colwidths-auto docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Name</p></th>
<th class="head"><p>Description</p></th>
<th class="head"><p>Usage</p></th>
<th class="head"><p>Default</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">-v</span></code>, <code class="docutils literal notranslate"><span class="pre">--verbose</span></code></p></td>
<td><p>Verbose logging flag</p></td>
<td><p>When enabled, count-alleles outputs a verbose log of the executiong</p></td>
<td><p>Not used</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">-g</span></code>, <code class="docutils literal notranslate"><span class="pre">--gamma</span></code></p></td>
<td><p>Level of confidence for selecting germline heterozygous SNPs</p></td>
<td><p>This value is the level of confidence used for the binomial model used to assess whether a called SNPs is in fact germline heterozygous</p></td>
<td><p>0.05</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">-q</span></code>, <code class="docutils literal notranslate"><span class="pre">--readquality</span></code></p></td>
<td><p>Threshold for phred-score quality of sequencing reads</p></td>
<td><p>The value can be either decreased (e.g. 10) or increased (e.g. 30) to adjust the filtering of sequencing reads</p></td>
<td><p>20</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">-Q</span></code>, <code class="docutils literal notranslate"><span class="pre">--basequality</span></code></p></td>
<td><p>Threshold for phred-score quality of sequenced nucleotide bases</p></td>
<td><p>The value can be either decreased (e.g. 10) or increased (e.g. 30) to adjust the filtering of sequenced nucleotide bases</p></td>
<td><p>20</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">-U</span></code>, <code class="docutils literal notranslate"><span class="pre">--snpquality</span></code></p></td>
<td><p>Threshold for phred-score quality of called variants</p></td>
<td><p>The value can be either decreased (e.g. 10) or increased (e.g. 30) to adjust the filtering of called variants</p></td>
<td><p>20</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">-L</span></code>, <code class="docutils literal notranslate"><span class="pre">--snps</span></code></p></td>
<td><p>Path to file of SNPs in the format <code class="docutils literal notranslate"><span class="pre">#CHR</span> <span class="pre">POS</span></code></p></td>
<td><p>When provided, only the included genomic positions will be considered for calling germline SNPs. Using well-known lists (e.g. dbSNP) help to significantly speed up this step</p></td>
<td><p>Not used, SNPs are called across all genome</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">-e</span></code>, <code class="docutils literal notranslate"><span class="pre">--regions</span></code></p></td>
<td><p>Path to file of regions in the format <code class="docutils literal notranslate"><span class="pre">#CHR</span> <span class="pre">START</span> <span class="pre">END</span></code> where <code class="docutils literal notranslate"><span class="pre">START</span></code> and <code class="docutils literal notranslate"><span class="pre">END</span></code> are the startning and ending positions in <code class="docutils literal notranslate"><span class="pre">CHR</span></code> for each region</p></td>
<td><p>When provided, germline SNPs are called only in the specified regions. This options is especially useful with WES data to specifify the exome regions</p></td>
<td><p>Not used, SNPs are called across all genome</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">-E</span></code>,<code class="docutils literal notranslate"><span class="pre">--newbaq</span></code></p></td>
<td><p>Flag to enable <code class="docutils literal notranslate"><span class="pre">newbaq</span></code> veafute of SAMtools</p></td>
<td><p>When selected, the user asks SAMtools to recompute alignment of reads on the fly during SNP calling</p></td>
<td><p>Not used</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">-b</span></code>, <code class="docutils literal notranslate"><span class="pre">--maxshift</span></code></p></td>
<td><p>Maximum BAF difference from 0.5</p></td>
<td><p>When used, only SNPs with an absolute difference between the BAF and 0.5 below the maximum are selected</p></td>
<td><p>Not used</p></td>
</tr>
</tbody>
</table>
</div>
</div>


           </div>
           
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>
        &#169; Copyright 2021, Princeton University.

    </p>
  </div>
    
    
    
    Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>
        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>