

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" />
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  
  <title>Tutorial &mdash; HATCHet 0.3.1 documentation</title>
  

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />

  
  

  
  

  

  
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="index.html" class="icon icon-home"> HATCHet
          

          
          </a>

          
            
            
              <div class="version">
                0.3.1
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul>
<li class="toctree-l1"><a class="reference internal" href="README.html">HATCHet</a></li>
</ul>
<ul>
<li class="toctree-l1"><a class="reference internal" href="modules.html">Modules</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">HATCHet</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          

















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html" class="icon icon-home"></a> &raquo;</li>
        
      <li>Tutorial</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
          
            <a href="_sources/doc_runhatchet.md.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="tutorial">
<h1>Tutorial<a class="headerlink" href="#tutorial" title="Permalink to this headline">¶</a></h1>
<p>This tutorial illustrates how to use the complete pipeline through the <code class="docutils literal notranslate"><span class="pre">hatchet</span> <span class="pre">run</span></code> command, along with the configuration <span class="xref myst">hatchet.ini</span> file that stores all the variables the user needs to specify.
The tutorial is subdivided into some subsections and each of these describes sequential parts of the full pipeline:</p>
<ol class="simple">
<li><p><a class="reference external" href="#preliminaries">Preliminaries</a></p></li>
<li><p><a class="reference external" href="#rundir">Setting up running directory</a></p></li>
<li><p><a class="reference external" href="#count-reads">count-reads</a></p></li>
<li><p><a class="reference external" href="#count-alleles">count-alleles</a></p></li>
<li><p><a class="reference external" href="#combine-counts">combine-counts</a></p></li>
<li><p><a class="reference external" href="#cluster-bins">cluster-bins</a></p></li>
<li><p><a class="reference external" href="#plot-bins">plot-bins</a></p></li>
<li><p><a class="reference external" href="#compute-cn">compute-cn</a></p></li>
<li><p><a class="reference external" href="#plot-cn">plot-cn</a></p></li>
</ol>
<p>We suggest you make a copy of the configuration file, place it into the designated running directory, and follow the tutorial.</p>
<div class="section" id="preliminaries">
<h2>Preliminaries<a class="headerlink" href="#preliminaries" title="Permalink to this headline">¶</a></h2>
<p><a name="preliminaries"></a></p>
<p>The following variables, specified in <span class="xref myst">hatchet.ini</span>, should be changed according to the user’s data:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span><span class="c1"># Path to reference genome - make sure you have also generated the reference dictionary as /path/to/reference.dict</span>
<span class="nv">reference</span> <span class="o">=</span> <span class="s2">&quot;/path/to/reference.fa&quot;</span>
<span class="nv">normal</span> <span class="o">=</span> <span class="s2">&quot;/path/to/normal.bam&quot;</span>
<span class="c1"># Space-delimited list of tumor BAM locations</span>
<span class="nv">bams</span> <span class="o">=</span> <span class="s2">&quot;/path/to/tumor1.bam /path/to/tumor2.bam&quot;</span>
<span class="c1"># Space-delimited list of tumor names</span>
<span class="nv">samples</span> <span class="o">=</span> <span class="s2">&quot;Primary Met&quot;</span>
</pre></div>
</div>
<p>Each of these are explained further below.</p>
<hr class="docutils" />
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span><span class="nv">reference</span> <span class="o">=</span> <span class="s2">&quot;/path/to/reference.fa&quot;</span>
<span class="nv">reference_version</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span>    
<span class="nv">chr_notation</span> <span class="o">=</span> True
</pre></div>
</div>
<p>First, you need to specify the full path to the human reference genome with the variable <code class="docutils literal notranslate"><span class="pre">reference</span></code>, along with the specific version of this reference with the <code class="docutils literal notranslate"><span class="pre">reference_version</span></code> variable, which may be “hg19” or “hg38”. This reference version is used to select a list of known germline SNPs to genotype samples. Lastly, the <code class="docutils literal notranslate"><span class="pre">chr_notation</span></code> must be set to true/false depending on whether or not the chromosomes in your reference are prefixed with “chr”.</p>
<hr class="docutils" />
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span><span class="nv">normal</span> <span class="o">=</span> <span class="s2">&quot;/path/to/normal.bam&quot;</span>
<span class="nv">bams</span> <span class="o">=</span> <span class="s2">&quot;/path/to/tumor1.bam /path/to/tumor2.bam&quot;</span>
<span class="nv">samples</span> <span class="o">=</span> <span class="s2">&quot;Primary Met&quot;</span>
</pre></div>
</div>
<p>Next, you need to specify the full paths to the required input data:</p>
<ol class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">normal</span></code> is the full path to the BAM file of matched-normal samples</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">bams</span></code> is a white-space separated list of the BAM files for the multiple tumor samples from the considered patient.</p></li>
</ol>
<p>The variable <code class="docutils literal notranslate"><span class="pre">samples</span></code> is also a white-space separated list of tumor sample names (specified in the same order as the BAM files in <code class="docutils literal notranslate"><span class="pre">bams</span></code>), and these names are used in the plots produced by HATCHet.</p>
<hr class="docutils" />
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span><span class="nv">mincov</span> <span class="o">=</span> <span class="m">8</span>
<span class="nv">maxcov</span> <span class="o">=</span> <span class="m">300</span>
<span class="nv">size</span> <span class="o">=</span> 50kb
</pre></div>
</div>
<p>Both <code class="docutils literal notranslate"><span class="pre">mincov</span></code> and <code class="docutils literal notranslate"><span class="pre">maxcov</span></code> specify the minimum and maximum sequencing depth for considering germline snps. For samples that have undergone whole-genome sequencing with depths &gt;30X, values around 8 and 300 should be reasonable for the minimum and maximum, respectively. For samples in which only the exome was sequenced to ~100X, then values around 20 and 1000 should be fine for the minimum and maximum, respectively.</p>
<p>The <code class="docutils literal notranslate"><span class="pre">size</span></code> variable indicates the size or genomic length of bins used to calculate the read-depth ratio (RDR) and B-Allele frequencies (BAF).</p>
</div>
<hr class="docutils" />
<div class="section" id="count-reads">
<h2>count-reads<a class="headerlink" href="#count-reads" title="Permalink to this headline">¶</a></h2>
<p><a name="count-reads"></a></p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>python3 -m hatchet count-reads -N <span class="si">${</span><span class="nv">NORMAL</span><span class="si">}</span> -T <span class="si">${</span><span class="nv">BAMS</span><span class="si">}</span> -S <span class="si">${</span><span class="nv">ALLNAMES</span><span class="si">}</span> -b <span class="si">${</span><span class="nv">BIN</span><span class="si">}</span> <span class="se">\</span>
                           -g <span class="si">${</span><span class="nv">REF</span><span class="si">}</span> -j <span class="si">${</span><span class="nv">J</span><span class="si">}</span> -O <span class="si">${</span><span class="nv">RDR</span><span class="si">}</span>normal.1bed -o <span class="si">${</span><span class="nv">RDR</span><span class="si">}</span>tumor.1bed <span class="se">\</span>
                           -t <span class="si">${</span><span class="nv">RDR</span><span class="si">}</span>total.tsv <span class="p">|&amp;</span> tee <span class="si">${</span><span class="nv">RDR</span><span class="si">}</span>bins.log
</pre></div>
</div>
<p><code class="docutils literal notranslate"><span class="pre">count-reads</span></code> computes the read counts in genomic bins of the reference genome from the matched-normal sample <code class="docutils literal notranslate"><span class="pre">${NORMAL}</span></code> and from all tumor samples <code class="docutils literal notranslate"><span class="pre">${BAMS}</span></code> with names <code class="docutils literal notranslate"><span class="pre">${ALLNAMES}</span></code>.
The size of genomic bins depend on the size of the CNAs that user aims to infer and on the noise in the data.
More specifically, shorter genomic bins allow to infer more refined CNAs, while larger genomic bins allow to better estimate the values of read-depth ratio (RDR) and B-allele frequency (BAF) for every bin.
The standard size 50kb typically represents a good compromise in whole-genome sequencing (WGS) data, while larger size (e.g. <code class="docutils literal notranslate"><span class="pre">200kb</span></code> or <code class="docutils literal notranslate"><span class="pre">250kb</span></code>) are better suited for whole-exome sequencing (WES) where we expect a sparser distribution of germline SNPs.
The reference genome <code class="docutils literal notranslate"><span class="pre">${REF}</span></code> is provided and <code class="docutils literal notranslate"><span class="pre">${REF}</span></code> must have a sequence dictionary file in the same folder, i.e. a file/dictionary with the same name but with a <code class="docutils literal notranslate"><span class="pre">.dict</span></code> extenion; this is used to know the length of the sequenced chromosomes.
Other simple parameters are also specified including number of parallel threads, output filenames, and the log filename of this step.</p>
</div>
<div class="section" id="genotype-snps">
<h2>genotype-snps<a class="headerlink" href="#genotype-snps" title="Permalink to this headline">¶</a></h2>
<p><a name="genotype-snps"></a></p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>python3 -m hatchet genotype-snps -N <span class="si">${</span><span class="nv">NORMAL</span><span class="si">}</span> -r <span class="si">${</span><span class="nv">REF</span><span class="si">}</span> -j <span class="si">${</span><span class="nv">J</span><span class="si">}</span> -c <span class="si">${</span><span class="nv">MINREADS</span><span class="si">}</span> -C <span class="si">${</span><span class="nv">MAXREADS</span><span class="si">}</span> <span class="se">\</span>
                            -R <span class="si">${</span><span class="nv">LIST</span><span class="si">}</span> -o <span class="si">${</span><span class="nv">SNP</span><span class="si">}</span> <span class="p">|&amp;</span> tee <span class="si">${</span><span class="nv">BAF</span><span class="si">}</span>bafs.log
</pre></div>
</div>
<p>genotype-snps genotypes germline SNPs from the matched-normal sample <code class="docutils literal notranslate"><span class="pre">${NORMAL}</span></code>, using positions of known germline variation specified by the file in <code class="docutils literal notranslate"><span class="pre">${LIST}</span></code>. If this parameter is omitted, heterozygous SNPs are inferred from the normal sample. As mentioned above, SNPs are only considered for downstream analysis if they have a minimum and maximum sequencing depth of <code class="docutils literal notranslate"><span class="pre">${MINREADS}</span></code> and <code class="docutils literal notranslate"><span class="pre">${MAXREADs}</span></code>, respectively.</p>
<p>GATK best practices suggest that the maximum should be at least twice the expected average coverage to avoid mapping artifacts.
Observe that WES generally requires higher thresholds (e.g. 100 and 3000).
Also, an increasing in the minimum threshold allow to improve the quality of the estimated BAF (computed in the next step), while a decreasing  it allows us to consider more SNPs.</p>
</div>
<div class="section" id="count-alleles">
<h2>count-alleles<a class="headerlink" href="#count-alleles" title="Permalink to this headline">¶</a></h2>
<p><a name="count-alleles"></a></p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>python3 -m hatchet count-alleles -N <span class="si">${</span><span class="nv">NORMAL</span><span class="si">}</span> -T <span class="si">${</span><span class="nv">BAMS</span><span class="si">}</span> -S <span class="si">${</span><span class="nv">ALLNAMES</span><span class="si">}</span> -r <span class="si">${</span><span class="nv">REF</span><span class="si">}</span> <span class="se">\</span>
                          -j <span class="si">${</span><span class="nv">J</span><span class="si">}</span> -c <span class="si">${</span><span class="nv">MINREADS</span><span class="si">}</span> -C <span class="si">${</span><span class="nv">MAXREADS</span><span class="si">}</span> -L <span class="si">${</span><span class="nv">SNP</span><span class="si">}</span>*.vcf.gz <span class="se">\</span>
                          -O <span class="si">${</span><span class="nv">BAF</span><span class="si">}</span>normal.1bed -o <span class="si">${</span><span class="nv">BAF</span><span class="si">}</span>tumor.1bed <span class="p">|&amp;</span> tee <span class="si">${</span><span class="nv">BAF</span><span class="si">}</span>bafs.log
</pre></div>
</div>
<p>count-alleles uses these germline SNPs from genotype-snps (specified in the list <code class="docutils literal notranslate"><span class="pre">${SNP}*.vcf.gz</span></code>) and computes the corresponding allele counts from all tumor samples <code class="docutils literal notranslate"><span class="pre">${BAMS}</span></code>.
As for all steps in this pipeline, the same reference genome used for the BAM files has to be specified by <code class="docutils literal notranslate"><span class="pre">${REF}</span></code>.
Minimum and maximum thresholds for the read counts of germline SNPs are again specified with <code class="docutils literal notranslate"><span class="pre">${MINREADS}</span></code> and <code class="docutils literal notranslate"><span class="pre">${MAXREADS}</span></code>.</p>
<p>Several simple parameters are also specified including: number of parallel threads, output filenames, and the log filename of this step.</p>
</div>
<div class="section" id="combine-counts">
<h2>combine-counts<a class="headerlink" href="#combine-counts" title="Permalink to this headline">¶</a></h2>
<p><a name="combine-counts"></a></p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>python3 -m hatchet combine-counts -c <span class="si">${</span><span class="nv">RDR</span><span class="si">}</span>normal.1bed -C <span class="si">${</span><span class="nv">RDR</span><span class="si">}</span>tumor.1bed -B <span class="si">${</span><span class="nv">BAF</span><span class="si">}</span>tumor.1bed <span class="se">\</span>
                          -t <span class="si">${</span><span class="nv">RDR</span><span class="si">}</span>total.tsv -p <span class="si">${</span><span class="nv">PHASE</span><span class="si">}</span> -l <span class="si">${</span><span class="nv">BLOCK</span><span class="si">}</span> -e <span class="si">${</span><span class="nv">RANDOM</span><span class="si">}</span> &gt; <span class="si">${</span><span class="nv">BB</span><span class="si">}</span>bulk.bb
</pre></div>
</div>
<p>combine-counts estimates the read-depth ratio (RDR) and B-Allele frequencies (BAF) from the read counts of all genomic bins from matched-normal sample in <code class="docutils literal notranslate"><span class="pre">${RDR}normal.1bed</span></code>, from the read counts of all genomic bins from all tumor samples in <code class="docutils literal notranslate"><span class="pre">${RDR}tumor.1bed</span></code>, and germline SNPs allele counts in <code class="docutils literal notranslate"><span class="pre">${BAF}tumor.1bed</span></code>. The variables <code class="docutils literal notranslate"><span class="pre">${PHASE}</span></code> and <code class="docutils literal notranslate"><span class="pre">${BLOCK}</span></code> are not described in detail in this tutorial (but see the <span class="xref myst">script</span> directory for a brief explanation of how to run HATCHet with phasing. Briefly, when <code class="docutils literal notranslate"><span class="pre">${PHASE}</span></code> specifies the location of a phased VCF file, HATCHet uses this VCF to combined phased SNPS within a haplotype block size of <code class="docutils literal notranslate"><span class="pre">${BLOCK}</span></code>.
The output from standard output is correspondingly written in a BB file <code class="docutils literal notranslate"><span class="pre">${BB}bulk.bb</span></code>.</p>
</div>
<div class="section" id="cluster-bins">
<h2>cluster-bins<a class="headerlink" href="#cluster-bins" title="Permalink to this headline">¶</a></h2>
<p><a name="cluster-bins"></a></p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>python3 -m hatchet cluster-bins <span class="si">${</span><span class="nv">BB</span><span class="si">}</span>bulk.bb -o <span class="si">${</span><span class="nv">BBC</span><span class="si">}</span>bulk.seg -O <span class="si">${</span><span class="nv">BBC</span><span class="si">}</span>bulk.bbc <span class="se">\</span>
                                      -e <span class="si">${</span><span class="nv">RANDOM</span><span class="si">}</span> -tB <span class="m">0</span>.04 -tR <span class="m">0</span>.15 -d <span class="m">0</span>.08
</pre></div>
</div>
<p>cluster-bins globally clusters genomic bins based on RDR and BAF jointly along the genome and across all tumor samples, specified in a BB file <code class="docutils literal notranslate"><span class="pre">${BB}bulk.bb</span></code>.
There are 2 main kind of parameters:</p>
<ul class="simple">
<li><p>The maximum expected BAF shift <code class="docutils literal notranslate"><span class="pre">-d</span></code> for diploid segments equal to <code class="docutils literal notranslate"><span class="pre">0.08</span></code>.</p></li>
<li><p>Thresholds for clustering refinement <code class="docutils literal notranslate"><span class="pre">-tB</span></code> and <code class="docutils literal notranslate"><span class="pre">-tR</span></code> are used to merge clusters whose difference is no more than these values in all samples.</p></li>
</ul>
<p>For all these parameters, these standard values are good for most of the datasets, however very noisy datasets may require higher thresholds.
In these cases, the values can be estimated and tuned by using the informative plots from plot-bins.
Other simple parameters allow to specify the output files containing the data of clusters in <code class="docutils literal notranslate"><span class="pre">${BBC}bulk.seg</span></code>, the one containing the clusterd genomic bins in <code class="docutils literal notranslate"><span class="pre">${BBC}bulk.bbc</span></code>.</p>
<p>This set up is generally effective when considering whole-genome sequencing (WGS) data.
However, a bootstraping approach allows to improve and empower the clustering when considering more sparse data as the ones from whole-exome sequeing (WES).
The bootsraping can be controlled by adding the following parameter:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>-u <span class="m">20</span> -dR <span class="m">0</span>.02 -dB <span class="m">0</span>.02
</pre></div>
</div>
<p>A total of <code class="docutils literal notranslate"><span class="pre">20</span></code> synthetic genomic bins are added only for the clustering by bootstraping each bin and generating RDR and BAF following normal distributions with <code class="docutils literal notranslate"><span class="pre">0.02</span></code> and <code class="docutils literal notranslate"><span class="pre">0.02</span></code>  variances, respectively. These values for <code class="docutils literal notranslate"><span class="pre">-dR</span></code> and <code class="docutils literal notranslate"><span class="pre">-dB</span></code> are specified by default, but the default value for <code class="docutils literal notranslate"><span class="pre">-u</span></code> is 0.</p>
</div>
<div class="section" id="plot-bins">
<h2>plot-bins<a class="headerlink" href="#plot-bins" title="Permalink to this headline">¶</a></h2>
<p><a name="plot-bins"></a></p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span><span class="nb">cd</span> <span class="si">${</span><span class="nv">PLO</span><span class="si">}</span>
python3 -m hatchet plot-bins -c RD --figsize <span class="m">6</span>,3 ../<span class="si">${</span><span class="nv">BBC</span><span class="si">}</span>bulk.bbc
python3 -m hatchet plot-bins -c CRD --figsize <span class="m">6</span>,3 ../<span class="si">${</span><span class="nv">BBC</span><span class="si">}</span>bulk.bbc
python3 -m hatchet plot-bins -c BAF --figsize <span class="m">6</span>,3 ../<span class="si">${</span><span class="nv">BBC</span><span class="si">}</span>bulk.bbc
python3 -m hatchet plot-bins -c BB ../<span class="si">${</span><span class="nv">BBC</span><span class="si">}</span>bulk.bbc
python3 -m hatchet plot-bins -c CBB ../<span class="si">${</span><span class="nv">BBC</span><span class="si">}</span>bulk.bbc -tS <span class="m">0</span>.01
</pre></div>
</div>
<p>plot-bins produces informative plots which are described <span class="xref myst">here</span>.
Many of these plots can be very useful to assess the performance of the various steps of HATCHet, especially in the case of noisy datasets.</p>
</div>
<div class="section" id="compute-cn">
<h2>compute-cn<a class="headerlink" href="#compute-cn" title="Permalink to this headline">¶</a></h2>
<p><a name="compute-cn"></a></p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span><span class="nb">cd</span> ../<span class="si">${</span><span class="nv">RES</span><span class="si">}</span>
python3 -m hatchet compute-cn -i ../<span class="si">${</span><span class="nv">BBC</span><span class="si">}</span>bulk -n2,6 -p <span class="m">400</span> -u <span class="m">0</span>.03 <span class="se">\</span>
                          -r <span class="si">${</span><span class="nv">RANDOM</span><span class="si">}</span> -j <span class="si">${</span><span class="nv">J</span><span class="si">}</span> -eD <span class="m">6</span> -eT <span class="m">12</span> -g <span class="m">0</span>.35 <span class="se">\</span>
                          -l <span class="m">0</span>.6 <span class="p">&amp;</span>&gt; &gt;<span class="o">(</span>tee &gt;<span class="o">(</span>grep -v Progress &gt; hatchet.log<span class="o">))</span>
</pre></div>
</div>
<p>compute-cn computes the fractional copy numbers, factorizes these into allele and clone-specific copy numbers and clone proportions, and applies a model-selection step to jointly infer the number of clones and predict a whole-genome duplication (WGD).
This step requires the common prefix <code class="docutils literal notranslate"><span class="pre">${BBC}bulk</span></code> of the cluster and clustered bins files.
Some basic parameter include:</p>
<ul class="simple">
<li><p>the interval <code class="docutils literal notranslate"><span class="pre">2,6</span></code>  specifying the minimum and maximum number of clones which can be increased as much as wanted. User can start to use this interval and increasing if the number of clones appear to be higher or close to the maximum.</p></li>
<li><p>the number of restarts <code class="docutils literal notranslate"><span class="pre">400</span></code> for the coordinate-descent method. This value is typically enough to obtain a good solution close to the optimum, however an increase in this value allows to improve the search of a solution. This can be useful when observing a number of clusters significantly higher than usual.</p></li>
<li><p>Random initializing seed with <code class="docutils literal notranslate"><span class="pre">${RANDOM}</span></code>.</p></li>
<li><p>Number of parallel threads <code class="docutils literal notranslate"><span class="pre">${J}</span></code></p></li>
</ul>
<p>There are also some important parameters which need to be tuned for special or noisy datasets.
The details of these parameters are the following:</p>
<ul class="simple">
<li><p>Maximum copy number without a WGD <code class="docutils literal notranslate"><span class="pre">-eD</span></code> and with a WGD <code class="docutils literal notranslate"><span class="pre">-eT</span></code>. The specified values are typically enough to obtain the copy numbers of large CNAs. While higher values increase the number of reasonable solutions, the values can be increased to investigate the presence of large genomic regions with higher copy numbers. In particular, when the value <code class="docutils literal notranslate"><span class="pre">0</span></code> is specified, then the maximum copy numbers are automatically chosen from the fractional copynumbers.</p></li>
<li><p>Minimum clone proportion <code class="docutils literal notranslate"><span class="pre">-u</span></code>. The value depends on the power to identify tumor clones present in samples at low tumor proportions. The default value is reasonable for most of the datasets, but datasets with high noise may need higher values to avoid overfitting. We thus suggest, especially in these special cases, to repeat the hatchet step by considering increasing values of this parameter up to a maximum or until clone proportions are different from this threshold.</p></li>
<li><p>Two parameters are used to control the elbow method during model selection to choose the best solution. The user should repeat this step with different values of these parameters, especially when one wants to investigate the presence of a single tumor clone or the presence of more tumor clones with small different CNAs and small clone proportions. These two parameters are the following:</p>
<ul>
<li><p>Confidence in the presence of a single tumor clone <code class="docutils literal notranslate"><span class="pre">g</span></code>. This value determines the confidence in having a single tumor clone in several of the given samples. The user can investigate the hypothesis that a single tumor clone is present by increasing the confidence, e.g. using values of <code class="docutils literal notranslate"><span class="pre">0.4,</span> <span class="pre">0.5,</span> <span class="pre">...</span></code>. Vice versa, the confidence is lowered by decreasing these values. For example, one can do that to investigate the presence of tumor clones with small CNAs or small proportions.</p></li>
<li><p>Sensitivity <code class="docutils literal notranslate"><span class="pre">l</span></code>. This value determines the sensitivity of the method. Lower values correspond to higher sensitivities which allow to consider solutions with a higher number of clones. The user can decrease this value for investigating the presence of more tumor clones, especially tumor clones with small different CNAs or present in small proportions.</p></li>
</ul>
</li>
</ul>
</div>
<div class="section" id="plot-cn">
<h2>plot-cn<a class="headerlink" href="#plot-cn" title="Permalink to this headline">¶</a></h2>
<p><a name="plot-cn"></a></p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span><span class="nb">cd</span> ../<span class="si">${</span><span class="nv">SUM</span><span class="si">}</span>
python3 -m hatchet plot-cn ../<span class="si">${</span><span class="nv">RES</span><span class="si">}</span>/best.bbc.ucn
</pre></div>
</div>
<p>plot-cn produces informative and summary plots for the inferred results.
Examples and details of these plots are reported <span class="xref myst">here</span>.</p>
</div>
</div>


           </div>
           
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>
        &#169; Copyright 2021, Princeton University.

    </p>
  </div>
    
    
    
    Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>
        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>