

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" />
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  
  <title>combine-counts &mdash; HATCHet 0.3.1 documentation</title>
  

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />

  
  

  
  

  

  
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="index.html" class="icon icon-home"> HATCHet
          

          
          </a>

          
            
            
              <div class="version">
                0.3.1
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul>
<li class="toctree-l1"><a class="reference internal" href="README.html">HATCHet</a></li>
</ul>
<ul>
<li class="toctree-l1"><a class="reference internal" href="modules.html">Modules</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">HATCHet</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          

















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html" class="icon icon-home"></a> &raquo;</li>
        
      <li>combine-counts</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
          
            <a href="_sources/doc_combine_counts.md.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="combine-counts">
<h1>combine-counts<a class="headerlink" href="#combine-counts" title="Permalink to this headline">¶</a></h1>
<p>This step combines the read counts and the allele counts for the identified germline SNPs to compute the read-depth ratio (RDR) and B-allele frequency (BAF) of every genomic bin.</p>
<div class="section" id="input">
<h2>Input<a class="headerlink" href="#input" title="Permalink to this headline">¶</a></h2>
<p>combine-counts takes in input three tab-separate files:</p>
<ol class="simple">
<li><p>A file of read counts for genomic bins obtained from the matched-normal sample, specified by the flag <code class="docutils literal notranslate"><span class="pre">-c</span></code>, <code class="docutils literal notranslate"><span class="pre">--normalbins</span></code>. The tab separated file has the following fields.</p></li>
</ol>
<table class="colwidths-auto docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Field</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">SAMPLE</span></code></p></td>
<td><p>Name of the matched-normal sample</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">CHR</span></code></p></td>
<td><p>Name of a chromosome</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">START</span></code></p></td>
<td><p>Starting genomic position of a genomic bin in <code class="docutils literal notranslate"><span class="pre">CHR</span></code></p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">END</span></code></p></td>
<td><p>Ending genomic position of a genomic bin in <code class="docutils literal notranslate"><span class="pre">CHR</span></code></p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">COUNT</span></code></p></td>
<td><p>Count of sequencing reads in the corresponding bin</p></td>
</tr>
</tbody>
</table>
<ol class="simple">
<li><p>A file of read counts for genomic bins obtained from all the tumor samples, specified by the flag <code class="docutils literal notranslate"><span class="pre">-C</span></code>, <code class="docutils literal notranslate"><span class="pre">--tumorbins</span></code>. The tab separated file has the following fields.</p></li>
</ol>
<table class="colwidths-auto docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Field</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">SAMPLE</span></code></p></td>
<td><p>Name of a tumor sample</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">CHR</span></code></p></td>
<td><p>Name of a chromosome</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">START</span></code></p></td>
<td><p>Starting genomic position of a genomic bin in <code class="docutils literal notranslate"><span class="pre">CHR</span></code></p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">END</span></code></p></td>
<td><p>Ending genomic position of a genomic bin in <code class="docutils literal notranslate"><span class="pre">CHR</span></code></p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">COUNT</span></code></p></td>
<td><p>Count of sequencing reads in the corresponding bin</p></td>
</tr>
</tbody>
</table>
<ol class="simple">
<li><p>A file of allele counts for heterozygous germline SNPs obtained from all the tumor samples, specified by the flag <code class="docutils literal notranslate"><span class="pre">-B</span></code>, <code class="docutils literal notranslate"><span class="pre">--tumorbafs</span></code>. The tab separated file has the following fields.</p></li>
</ol>
<table class="colwidths-auto docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Field</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">SAMPLE</span></code></p></td>
<td><p>Name of a tumor sample</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">CHR</span></code></p></td>
<td><p>Name of a chromosome</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">POS</span></code></p></td>
<td><p>Genomic position corresponding to a heterozygous germline in <code class="docutils literal notranslate"><span class="pre">CHR</span></code></p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">REF_COUNT</span></code></p></td>
<td><p>Count of reads covering <code class="docutils literal notranslate"><span class="pre">POS</span></code> with reference allele</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">ALT_COUNT</span></code></p></td>
<td><p>Count of reads covering <code class="docutils literal notranslate"><span class="pre">POS</span></code> with alternate allele</p></td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="output">
<h2>Output<a class="headerlink" href="#output" title="Permalink to this headline">¶</a></h2>
<p>combine-counts produces a tab-separated file with the following fields.</p>
<table class="colwidths-auto docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Field</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">CHR</span></code></p></td>
<td><p>Name of a chromosome</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">START</span></code></p></td>
<td><p>Starting genomic position of a genomic bin in <code class="docutils literal notranslate"><span class="pre">CHR</span></code></p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">END</span></code></p></td>
<td><p>Ending genomic position of a genomic bin in <code class="docutils literal notranslate"><span class="pre">CHR</span></code></p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">SAMPLE</span></code></p></td>
<td><p>Name of a tumor sample</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">RD</span></code></p></td>
<td><p>RDR of the bin in <code class="docutils literal notranslate"><span class="pre">SAMPLE</span></code></p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">#SNPS</span></code></p></td>
<td><p>Number of SNPs present in the bin in <code class="docutils literal notranslate"><span class="pre">SAMPLE</span></code></p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">COV</span></code></p></td>
<td><p>Average coverage in the bin in <code class="docutils literal notranslate"><span class="pre">SAMPLE</span></code></p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">ALPHA</span></code></p></td>
<td><p>Alpha parameter related to the binomial model of BAF for the bin in <code class="docutils literal notranslate"><span class="pre">SAMPLE</span></code>, typically total number of reads from A allele</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">BETA</span></code></p></td>
<td><p>Beta parameter related to the binomial model of BAF for the bin in <code class="docutils literal notranslate"><span class="pre">SAMPLE</span></code>, typically total number of reads from B allele</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">BAF</span></code></p></td>
<td><p>BAF of the bin in <code class="docutils literal notranslate"><span class="pre">SAMPLE</span></code></p></td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="main-parameters">
<h2>Main parameters<a class="headerlink" href="#main-parameters" title="Permalink to this headline">¶</a></h2>
<p>combine-counts has some main parameters; the main values of these parameters allow to deal with most of datasets, but their values can be changed or tuned to accommodate the features of special datasets.</p>
<table class="colwidths-auto docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Name</p></th>
<th class="head"><p>Description</p></th>
<th class="head"><p>Usage</p></th>
<th class="head"><p>Default</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">-d</span></code>, <code class="docutils literal notranslate"><span class="pre">--diploidbaf</span></code></p></td>
<td><p>Maximum expected shift from 0.5 for BAF of diploid or tetraploid clusters</p></td>
<td><p>The maximum shift is used to identify all the potential clusters with base states (1, 1) or (2, 2). The value depends on the variance in the data (related to noise and coverage); generally, higher variance requires a higher shift. Information provided by plot-bins can help to decide this value in special datasets.</p></td>
<td><p>0.08 (other typically suggested values are 0.1-0.11 for higher variance and 0.06 for low variance)</p></td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="optional-parameters">
<h2>Optional parameters<a class="headerlink" href="#optional-parameters" title="Permalink to this headline">¶</a></h2>
<table class="colwidths-auto docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Name</p></th>
<th class="head"><p>Description</p></th>
<th class="head"><p>Usage</p></th>
<th class="head"><p>Default</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">-v</span></code>, <code class="docutils literal notranslate"><span class="pre">--verbose</span></code></p></td>
<td><p>Verbose logging flag</p></td>
<td><p>When enabled, combine-counts outputs a verbose log of the executiong</p></td>
<td><p>Not used</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">-r</span></code>, <code class="docutils literal notranslate"><span class="pre">--disablebar</span></code></p></td>
<td><p>Disabling progress-bar flag</p></td>
<td><p>When enabled, the output progress bar is disabled</p></td>
<td><p>Not used</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">-b</span></code>, <code class="docutils literal notranslate"><span class="pre">--normalbafs</span></code></p></td>
<td><p>File of allele counts for SNPs in matched-normal sample</p></td>
<td><p>When provided, combine-counts attempts to correct the estimated BAF using the variance in matched-normal sample.</p></td>
<td><p>Not used (deprecated)</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">-d</span></code>, <code class="docutils literal notranslate"><span class="pre">--diploidbaf</span></code></p></td>
<td><p>Maximum expected shift from 0.5 for BAF of diploid or tetraploid clusters</p></td>
<td><p>The maximum shift is used to identify potential potential bins with base states (1, 1) or (2, 2) whose BAF needs to be corrected. The value depends on the variance in the data (related to noise and coverage); generally, higher variance requires a higher shift. Information provided by plot-bins can help to decide this value in special datasets.</p></td>
<td><p>0.08 (other typically suggested values are 0.1-0.11 for higher variance and 0.06 for low variance)</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">-t</span></code>, <code class="docutils literal notranslate"><span class="pre">--totalcounts</span></code></p></td>
<td><p>File of total read counts</p></td>
<td><p>When provided, the total read counts are used to normalize the read counts from the corresponding sample</p></td>
<td><p>Not used (deprecated)</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">-m</span></code>, <code class="docutils literal notranslate"><span class="pre">--mode</span></code></p></td>
<td><p>Mode used to estimate BAFs</p></td>
<td><p>Different modes are provided to combine to combine the allele counts of SNPs</p></td>
<td><p>The counts from the allele with minor count are combined (deprecated)</p></td>
</tr>
</tbody>
</table>
</div>
</div>


           </div>
           
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>
        &#169; Copyright 2021, Princeton University.

    </p>
  </div>
    
    
    
    Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>
        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>